plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -266,8 +275,41 @@ private Client client() {
     @FunctionalInterface
     interface ProcessorFactoryProvider {
 
-        Processor.Factory get(Environment environment);
+        Processor.Factory get(Environment environment, TemplateService templateService);
+
+    }
+
+    static class InternalTemplateService implements TemplateService {
```

Comment:
Agreed, so lets move some inner classes to top level ones? Like `ProcessorFactoryProvider` and `InternalTemplateService `?